packages/web-service/src/pages/common/yes-no-conditional.js

```diff
@@ -1,34 +1,36 @@
 import Joi from 'joi'
 import pageRoute from '../../routes/page-route.js'
 
-export const validator = async payload => {
+export const validator = async (payload, skipValidations) => {
   if (!payload['yes-no']) {
     Joi.assert(payload, Joi.object({
       'yes-no': Joi.any().required()
     }).options({ abortEarly: false, allowUnknown: true }))
   }
 
-  if (payload['yes-no'] === 'yes') {
+  if (payload['yes-no'] === 'yes' && (!skipValidations || (skipValidations && !skipValidations.includes('yes-conditional-input')))) {
```

Comment:
If we wanted we could shorten the condition a little with optional chaining:
```suggestion
  if (payload['yes-no'] === 'yes' && !skipValidations?.includes('yes-conditional-input')) {
```

Reply:
I'm afraid the logic isn't the same here. What I was going for was: 

- if the Yes/No field is 'yes' AND either (skipValidations is undefined OR skipValidations is set AND contains "yes-conditional-input")

Whereas the suggestion if applied would change it to: 

- if the Yes/No field is 'yes' AND skipValidations is set AND contains "yes-conditional-input"

You were right to see that it can be shortened though... This will produce the result I'm looking for as the right side of the OR assumes it has a value. Its also more explicit about the checking if the value is undefined, rather than just falsy.
```
if (payload['yes-no'] === 'yes' && (skipValidations === undefined || !skipValidations.includes('yes-conditional-input')))
```

Reply:
I quickly plugged it into a REPL and the suggested change does do what we want:

* It gives us `true` if `skipValidations` is set and doesn't include `yes-conditional-input`;
* It gives us `false` if `skipValidations` is set and does include `yes-conditional-input`;
* It gives us `true` if `skipValidations` isn't set.

But happy to go with `skipValidations === undefined || !skipValidations.includes('yes-conditional-input')` -- it does make the logic of what we're checking more explicit 😁

Reply:
Ah ok. Glad you checked it. Good to know the exact behaviour of the optional chaining in this case. 👍